codegen/model/chart.go

```diff
@@ -50,24 +55,18 @@ type Operator struct {
 
 // values for Deployment template
 type Deployment struct {
-	// use a DaemonSet instead of a Deployment
-	UseDaemonSet               bool                     `json:"-"`
-	ExtraPodLabels             map[string]string        `json:"extraPodLabels,omitempty"`
-	ExtraPodAnnotations        map[string]string        `json:"extraPodAnnotations,omitempty"`
-	Image                      Image                    `json:"image,omitempty"`
-	ExtraDeploymentLabels      map[string]string        `json:"extraDeploymentLabels,omitempty"`
-	ExtraDeploymentAnnotations map[string]string        `json:"extraDeploymentAnnotations,omitempty"`
-	DeploymentImage            Image                    `json:"image,omitempty"`
-	Resources                  *v1.ResourceRequirements `json:"resources,omitempty"`
+	// TODO support use of a DaemonSet instead of a Deployment
+	UseDaemonSet bool                     `json:"-"`
+	Image        Image                    `json:"image,omitempty"`
+	Resources    *v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
We will need a Deployment Overrides section too

Reply:
i don't believe we'll need it here. Deployment overrides are meant to be exposed directly to the user so they can override what we ship in the final chart. the chart itself is something we generate based on the `Operator` type defined in this file. 

So if an skv2 project needs to set some properties on the deployment, it can just configure the `Operator` type being passed to skv2. we can discuss modifying skv2 to expose more aspects of the deployment, but for now i think that's out of scope for this change.

Reply:
Sorry, that was super unclear - we need a way to modify the default Helm Values (e.g. by adding a `sidecar.istio.io/inject : false` annotation)